### DISCUSSION TITLE
fbx animation feature support

### DISCUSSION THREAD

**Discussion by ggslayer** [vsg-dev/VulkanSceneGraph#999] [General]
Hi, I asked a question about amimation support by vsg,
In osg issue (https://github.com/openscenegraph/OpenSceneGraph/issues/1258),  where I know the support will done by vsg in this year,  In the sketlon animation ability, I want some advance features, like multi animation play concurrent、 animations mix,  animation layer overlay,   for example: 
https://esotericsoftware.com/spine-applying-animations
https://www.cnblogs.com/BigFeng/p/6781855.html,  
the pages is spine2d animation's feature, these feature is import for implements expected smoothly vision affect.   

I want to know is these feature will be considered and added?   thank you

**Comment by robertosfield** [comment]
I haven't looked into the design details or specific features for implementing animation.  Rather than look for a list of must have features I find it useful to create usage cases that we wish to solve and then figure out what features are required to address them.

One area that can help give shape what we might attempt first would be to find some public datasets that have rigid body animation and ones that have skinning that we want to be able to load and animate.  Rigid body animation would likely be implemented on the CPU via transform nodes, while skinning would be done on as part of the VSG's built-in vertex shaders,

The vsgXchange::assimp loader would likely be the first area where we'd want to pull in animation data from, and I have a whole back I briefly looked at the data structures that assimp provides so I am aware that they are available, but at the time I was focused on other functionality so didn't dive any deeper.

As I don't want to spread myself too thinly across too many concurrent topics I don't want to dive into the details of specific features and implementations right now.  My current focus will be getting the new shadow mapping functionality wrapped up and review/merge PRs, address reported issues and then make the next stable release of the VSG.  At current rate of work this will probably be first half to mid November.

After the next stable release I will tackle new topics.  Essentially bit by bit the VSG project is tackling features that the OSG and other projects have, and when doing so implement things in modern ways so once the new implementation comes online the feature ends up being easier to use, better quality and more efficient than what came before it.

For members of the community that want to see animation support please chip into this thread, concrete examples/usage cases are generally the most helpful, as are 3rd party examples of how data is loaded, what shaders are used etc.  If there are Vulkan related examples illustrating what is required then all the better.

**Comment by robertosfield** [comment]
I forgot to mention that the vsg::AnimationPath class exists for representing rigid body paths, this is equivalent to the osg::AnimationPath.  

The main use of the [vsg::AnimationPath](https://github.com/vsg-dev/VulkanSceneGraph/blob/master/include/vsg/utils/AnimationPath.h#L24) is currently animation the vsg::Camera path, and is something I use regularly in testing, especially performance testing.  You can even load the OSG .path files thanks to support vsgXchange.  

There is also a [vsg::AnimationPathHandler](https://github.com/vsg-dev/VulkanSceneGraph/blob/master/include/vsg/utils/AnimationPath.h#L61) for replaying an animation path, and just added is a new [vsg::RecordAnimationPathHandler](https://github.com/vsg-dev/VulkanSceneGraph/blob/master/include/vsg/utils/AnimationPath.h#L99) classes that can both record the path that a view's Camera is following, write the path out to file, and replay it.

**Comment by robertosfield** [comment]
To provide a bit of closure on this old thread, I have now checked animation support and written up details on this thread:

https://github.com/vsg-dev/VulkanSceneGraph/discussions/1095
